src/System.Runtime.Extensions/src/System/IO/Path.Windows.cs

```diff
@@ -176,16 +176,7 @@ private unsafe static string NormalizePath(string path, bool fullCheck, int maxP
             // since StringBuilder is used. 
             // 2. IsolatedStorage, which supports paths longer than MaxPath (value given 
             // by maxPathLength.
-            PathHelper newBuffer = null;
-            if (path.Length + 1 <= MaxPath)
-            {
-                char* m_arrayPtr = stackalloc char[MaxPath];
-                newBuffer = new PathHelper(m_arrayPtr, MaxPath);
-            }
-            else
-            {
-                newBuffer = new PathHelper(path.Length + MaxPath, maxPathLength);
-            }
+            PathHelper newBuffer = new PathHelper(path.Length + MaxPath, maxPathLength);
```

Comment:
Isn't there a known maximum amount by which it will need to grow, measured in just a few characters?  If so, couldn't we just proactively make the stackalloc'd size that much larger?